docker/Dockerfile.multiplexer

```diff
@@ -0,0 +1,99 @@
+# This Dockerfile performs a multi-stage build. BUILDER_IMAGE is the image used
+# to compile the celestia-appd binary. RUNTIME_IMAGE is the image that will be
+# returned with the final celestia-appd binary.
+#
+# Separating the builder and runtime image allows the runtime image to be
+# considerably smaller because it doesn't need to have Golang installed.
+ARG BUILDER_IMAGE=docker.io/golang:1.23.6-alpine3.20
+ARG RUNTIME_IMAGE=docker.io/alpine:3.19
+ARG TARGETOS
+ARG TARGETARCH
+# Use build args to override the maximum square size of the docker image e.g.
+# docker build --build-arg MAX_SQUARE_SIZE=64 -t celestia-app:latest .
+ARG MAX_SQUARE_SIZE
+# Use build args to override the upgrade height delay of the docker image e.g.
+# docker build --build-arg UPGRADE_HEIGHT_DELAY=1000 -t celestia-app:latest .
+ARG UPGRADE_HEIGHT_DELAY
+# the tag used for the embedded v3 binary.
+ARG CELESTIA_VERSION=v3.4.2
+# the docker registry used for the embedded v3 binary.
+ARG CELESTIA_APP_REPOSITORY=ghcr.io/celestiaorg/celestia-app
+
+# Stage 1: this base image contains already released binaries which can be embedded in the multiplexer.
+FROM ${CELESTIA_APP_REPOSITORY}:${CELESTIA_VERSION} AS base
+
+# Stage 2: Build the celestia-appd binary inside a builder image that will be discarded later.
+# Ignore hadolint rule because hadolint can't parse the variable.
+# See https://github.com/hadolint/hadolint/issues/339
+# hadolint ignore=DL3006
+FROM --platform=$BUILDPLATFORM ${BUILDER_IMAGE} AS builder
+ENV CGO_ENABLED=0
+ENV GO111MODULE=on
+# hadolint ignore=DL3018
+RUN apk update && apk add --no-cache \
+    gcc \
+    git \
+    # linux-headers are needed for Ledger support
+    linux-headers \
+    make \
+    musl-dev
+
+WORKDIR /celestia-app
+
+# cache go module dependencies
+COPY go.mod go.sum ./
+RUN go mod download
+
+# copy source code after downloading modules (to leverage caching)
+COPY . .
```

Comment:
this should be updated to copy only the files required. left as is as this is how it is in the regular Dockerfile. I can open a follow up PR to update this for both images and verify they still run okay.